src/front/glsl_new/ast.rs

```diff
@@ -1,28 +1,30 @@
-use crate::{Arena, Constant, FastHashMap, Function, Handle, Type};
+use crate::{Arena, Constant, FastHashMap, Function, GlobalVariable, Handle, ShaderStage, Type};
 
 #[derive(Debug)]
 pub struct Program {
     pub version: u16,
     pub profile: Profile,
-    pub ext_decls: Vec<ExtDecl>,
+    pub shader_stage: ShaderStage,
     pub lookup_function: FastHashMap<String, Handle<Function>>,
     pub functions: Arena<Function>,
     pub lookup_type: FastHashMap<String, Handle<Type>>,
     pub types: Arena<Type>,
     pub constants: Arena<Constant>,
+    pub global_variables: Arena<GlobalVariable>,
 }
 
 impl Program {
     pub fn new() -> Program {
         Program {
             version: 0,
             profile: Profile::Core,
-            ext_decls: vec![],
+            shader_stage: ShaderStage::Vertex,
```

Comment:
why is Vertex the default? Let it be a `new` argument instead

Reply:
Done :)